### ISSUE TITLE
Cypress erasing cookies during test when running on Chrome browser

### ISSUE THREAD

**Issue by Zach-Costa** [cypress-io/cypress#17045]
Recently, my org has made an overhaul to our SAML Identity Provider for internal users. Before now, we were using a hybrid of Auth0 and OneLogin to access our internal app, but we've now moved to a hybrid of Auth0 and Okta. Our specs have always logged in via the UI with no issues, and we were able to log in through our new system via UI without any problems. After a couple of weeks, however, our specs began to encounter an issue where the session token would be removed after logging in, redirecting the test to our "Not Logged in" landing page. 

### Expected behavior

Here's how our current system works:

1. User tries to access app without an active session
2. Auth0 makes a request to Okta for authentication
3. Okta redirects the user to the Okta login page
4. User submits a valid username, password, and passcode
5. Okta sets a session token for the user and forwards a SAML response to Auth0
6. Auth0 sends a JWT to a callback URL in our Rails application, which validates the token, records some data from the token into the user's session, and redirects the user
7. The app reads the token data out of the session and signs the user into our system

### Current behavior

When running Cypress on Electron, Firefox, or in Headless mode, this works exactly as described above. However, when running in Chrome, the session cookie gets dropped between steps 6 and 7, resulting in the spec failing. This ONLY occurs when running tests against the Chrome browser, and it also ONLY occurs with Cypress.

When logging the Cookie changes with `Cypress.Cookies.debug()`, nothing is reported when the session token gets dropped. However, we've been able to confirm that this issue is Cypress-specific, and occurs consistently 100% of the time. As Chrome is the browser we've always recommended users use with our app, it's been a huge blocker and time sink not being able to run tests on the main browser we support.

I'm aware that Cypress clears out the cookies between tests, but this issue is happening in the middle of the `beforeEach()` step, which shouldn't happen.

### Fixes already attempted

- Adding in `Cypress.Cookies.preserveOnce("session_token")` does not work, the session token still gets dropped
- We're still investigating logging in via the API, but since we actually do need to test the login flow that our end-users will go through, this would only be a solution for some of our tests. We need to be able to login via the UI to make sure we're testing our login flow as well. In addition, the Okta API token gives more permissions than our security team is comfortable with giving to a large group of people
- We've run our tests in secure Docker Image to remove any discrepancies caused by our local environments. We've also attempted setting up a brand new Cypress repo to test against. In both instances, the bug occurred.
- Tried programmatically pulling the session ID after it's set, and then resetting it after the cookie is dropped. This resulted in an unusual state where the user became logged into Okta, but Auth0 would no longer recognize the token.

### Versions

This has been tested on several builds, including 7.2.0, 6.3.0, and 5.2.0. The bug seems to be version-agnostic.

**Comment by jennifer-shehane** [comment]
Since you mention this used to work fine in Chrome, I would try to track down what changed between the working version and broken version. You may want to try running on different Chrome versions to see if something changed in a specific Chrome version.

Also make sure to check the DevTools for any warnings in Chrome - may be they are handling something differently. 

There was an interesting case where the state of the application could 'leak' into the next 'beforeEach' described here that may be relevant: https://github.com/cypress-io/cypress/issues/16935 Although I'm not sure why this would be Chrome specific, so maybe not.

**Comment by Zach-Costa** [comment]
The issue isn't related to the Chrome version, as this issue began after we migrated our system to Okta. Reverting my Chrome version doesn't impact the issue either, so it's definitely something to do with Cypress, Chrome, and Okta interacting weirdly. Chrome is not reporting any errors either, the cookie is just randomly disappearing.

The issue you linked describes some similar behaviour - But it also states that the it's an issue of the first test contaminating later tests. In our case, the first test never even manages to log in.

**Comment by Zach-Costa** [comment]
@jennifer-shehane Would you be able to remove the "awaiting-response" label?

**Comment by jennifer-shehane** [comment]
These types of issues are really difficult to begin debugging since it's so specific to the application under test. It's not obvious from the information given on why this would be happening.  Nothing also really stands out as something previously reported for [cookie issues](https://github.com/cypress-io/cypress/issues?page=1&q=is%3Aissue+is%3Aopen+cookies+label%3A%22topic%3A+cookies+%3Acookie%3A%22).

If you can provide a way to reproduce this or narrow down the circumstances where a cookie would disappear/not be set, we can try to take a look at it.

**Comment by Zach-Costa** [comment]
Actually, found a solution to this! The issue, [as described here](https://github.com/cypress-io/cypress/issues/16329), is the SameSite Cookies feature Chrome rolled out last year. Disabling that, per the solution in the link, fixes the problem entirely. My question then, is why did Cypress not begin encountering this until several weeks after we had switched to Okta? Theoretically, since Cypress clears out the cache between runs, we should have begun encountering this bug as soon as we switched to Okta, but that wasn't the case.
